### ISSUE TITLE
Will wtpoa-cns automatically exit after the run is complete?

### ISSUE THREAD

**Issue by YangJianshun** [ruanjue/wtdbg2#50]
I run the command “./wtpoa-cns -t 16 -i prefix.ctg.lay -fo prefix.ctg.lay.fa”, but after generating the prefix.ctg.lay.fa file for a long time, wtpoa-cns does not Automatic termination, is this normal?

**Comment by ruanjue** [comment]
Abnormal. What's about the log information? Could you copy the text here.

**Comment by YangJianshun** [comment]
> Abnormal. What's about the log information? Could you copy the text here.

$ /public/home/yangjs/software/wtdbg2/wtpoa-cns -t 16 -i BAC_nanopore.ctg.lay.gz -fo BAC_nanopore.final.fa
--
-- total memory       32727680.0 kB
-- available          30977672.0 kB
-- 4 cores
-- Starting program: /public/home/yangjs/software/wtdbg2/wtpoa-cns -t 16 -i BAC_nanopore.ctg.lay.gz -fo BAC_nanopore.final.fa
-- pid                     31901
-- date         Thu Dec  6 13:59:36 2018
--
11 contigs 800 edges


No longer change and don't quit
But I did get this file ‘’BAC_nanopore.final.fa’’

**Comment by ruanjue** [comment]
Thanks Jianshun!

Please try new commit https://github.com/ruanjue/wtdbg2/commit/31cf61b4dfb1c3fdae56a6d9a48fa77442944f3f

Best,
Jue

**Comment by YangJianshun** [comment]
> Thanks Jianshun!
> 
> Please try new commit [31cf61b](https://github.com/ruanjue/wtdbg2/commit/31cf61b4dfb1c3fdae56a6d9a48fa77442944f3f)
> 
> Best,
> Jue

Thank you very much

**Comment by zrlewis** [comment]
Hi,

Jumping on this closed issue here because I think that I'm seeing the same behavior.

`wtpoa-cns` appears to hang and not finish. The end of the log file lists that it has gone through all of the contigs and edges, but the program does not exit. 

I'm working off of the most recent commit 

```
git show --oneline -s
5009cff fixed a bug in option -K
```

My command was:

`wtpoa-cns -t 20 -i  [filename].ctg.lay.gz -o [filename].ctg.lay.fa`

***
For reference, my `wtdbg2` command was:

```
wtdbg2 -t 20 -i concatenated_reads.fasta \
   -fo output_wtdbg2 \
   -p 0 -k 15 -AS 2 -s 0.05 -L 8000 \
   -g 3.1g \
   --edge-min 2 --rescue-low-cov-edges --aln-noskip \
   --ttr-cutoff-depth 20 \
   -K 1000.0
```


Here is the tail of the `wtdbg2` output:

```
[Thu Dec  6 16:46:58 2018] building contigs
[Thu Dec  6 16:46:59 2018] searched 120585 contigs
[Thu Dec  6 16:46:59 2018] Estimated: TOT 2910323968, CNT 50730, AVG 57369, MAX 1501696, N50 165376, L50 4860, N90 21760, L90 23422, Min 5120
[Thu Dec  6 16:54:46 2018] output 50730 contigs
[Thu Dec  6 16:54:52 2018] Program Done
** PROC_STAT(TOTAL) **: real 106937.848 sec, user 2052869.090 sec, sys 25841.980 sec, maxrss 252911340.0 kB, maxvsize 279355532.0 kB
---
```

**Comment by ruanjue** [comment]
Dear zrlewis,

I cannot locate the bug from reading the code, for I haven't find a test data reproduce this problem.

If anyone find a small test data, I will continue to work on it.

Best,
Jue

**Comment by zrlewis** [comment]
@ruanjue Thanks for looking into it. I'm trying a couple new assemblies from scratch using the most recent commit and will let you know if I still encounter the problem.

**Comment by zrlewis** [comment]
@ruanjue `wtpoa-cns` finished this time after running both `wtdbg2` and consensus from the most recent commit. I had previously been running a slightly older `wtdbg2` version. It was labeled `Version: 2.2 (20181111)`, but I don't have the commit. That could have been the problem.

**Comment by ruanjue** [comment]
I believe there is still similar problem in wtpoa-cns, but have no idea where to find it. Let's talk about it when it re-occuring.

Jue

**Comment by shri1984** [comment]
Hi, 
I am using latest copy of wtdbg2 from github. I also see a similar issue. wtpoa-cns reaches very end and then don't exit. i dont know whether this issue is related to the parameters I used in wtdbg2 run. I never issues before when I ran wtdbg2 with default settings.

**Comment by ruanjue** [comment]
It looked like some contig took very long time or endless. Try to split the input fle into multiple ones, and run wtpoa-cns seperately.

**Comment by shri1984** [comment]
Thanks. You are right, there is one contig look endless.  
I managed to split the ctg.lay file into multiple files and manage to finish them independently. However, when I counted the number of contigs in ctg.lay and concatenated (from independent files) .raw.fa. I observed a difference of 1 contig, and I isolated that contig which is not showing up in the final raw. fa and run it wtpoa-cns.  It does not look like finishing any time soon. It is running for 2 hours for that one contig. any input?
